readme.md

```diff
@@ -991,6 +992,32 @@ Default: `false`
 
 A read-only boolean describing whether the defaults are mutable or not. If set to `true`, you can [update headers over time](#hooksafterresponse), for example, update an access token when it expires.
 
+## Types
+
+When used in a TypeScript project, Got has a set of handy types and interfaces available.
+
+### Got
+
+Typescript will automatically infer types for Got instances but in case you want to define something like dependencies you can use following: `Got`, `GotStream`, `ReturnStream`, `GotRequestMethod`, `GotReturn`
```

Comment:
I'm not sure the value of just listing these types. The user has no idea what they do or what to use them for. Maybe better to just mention one type with example and refer to the type definition instead. It's also annoying to have to keep this in sync with the source.

Same with hooks.

Reply:
I was thinking exactly the same when I was writing it, but I had to start somewhere. I'll try to shorten things up